### PR TITLE
CORE-3977 bbox accuracy in hi-res

### PR DESCRIPTION
**Pull request by amaciaszek-dsai**
The script is used to evaluate the hi-res model outputs using the labels provided in the labels_path.

* hi-res model outputs can be downloaded from S3:
_s3://utic-comparative-metrics/mini-holistic/unstructured-core-hi_res/_

* Labels can be downloaded from DVC:
_dvc-data-registry/holistic-mini-pdf-image-dataset/mini-holistic-all/ls/export_45956_project-45956-at-2024-01-10-23-16-24cfbda6.json_

To run the script, use the following command:
`python helper_scripts/evaluate_hi_res.py --output_dir <path_to_model_outputs> --labels_path <path_to_labels> [--default_detection_class_prob 1.0]`

on deepsense pascal environment that would be:
`python helper_scripts/evaluate_hi_res.py --output_dir /mnt/ml-team/unstructured/adam/unstructured-core-hi_res/json --labels_path /mnt/ml-team/unstructured/DATA/dvc-data-registry/holistic-mini-pdf-image-dataset/mini-holistic-all/ls/export_45956_project-45956-at-2024-01-10-23-16-24cfbda6.json`

### CODE REVIEW

File: helper_scripts/evaluate_hi_res.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+dotenv.load_dotenv()
+logging.basicConfig(level=logging.INFO, format="%(levelname)s: %(message)s")
+logger = logging.getLogger(__name__)
+
+DEFAULT_DETECTION_CLASS_PROB = 1.0
+
+NEPTUNE_PARAMS = {
+    "name": "hi_res_evaluation",
+    "tags": ["hi_res", "eval", "CORE-3977"],
```
Comment (mariannaparzych) [inline helper_scripts/evaluate_hi_res.py:39]: maybe task name should be <task_id> - I assume script will be also used for future evaluations, in diffenet tasks

File: helper_scripts/evaluate_hi_res.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    return img_width or label_width or 0, img_height or label_height or 0
+
+
+def main(output_dir: Path, labels_path: Path, default_detection_class_prob: float):
+    neptune_run = neptune.init_run(**NEPTUNE_PARAMS)
+    logger.info(f"Neptune run created: {neptune_run}")
+
+    device = "cpu"
+
+    with open(labels_path, "r") as file:
```
Comment (mariannaparzych) [inline helper_scripts/evaluate_hi_res.py:138]: we have load_json in utils

File: helper_scripts/evaluate_hi_res.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+    document_labels = {}
+
+    for document_filename in os.listdir(output_dir):
+        logger.info(f"Processing {document_filename}")
+        document_name = Path(document_filename).stem
+        document_labels[document_name] = {}
+
+        if document_filename.endswith(".json"):
+            with open(output_dir / document_filename, "r") as file:
```
Comment (mariannaparzych) [inline helper_scripts/evaluate_hi_res.py:150]: j/w

File: helper_scripts/mappings.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    {"supercategory": "Page-footer", "id": 5, "name": "Page-footer"},
+    {"supercategory": "Page-header", "id": 6, "name": "Page-header"},
+    {"supercategory": "Picture", "id": 7, "name": "Picture"},
+    {"supercategory": "Section-header", "id": 8, "name": "Section-header"},
+    {"supercategory": "Table", "id": 9, "name": "Table"},
+    {"supercategory": "Text", "id": 10, "name": "Text"},
+    {"supercategory": "Title", "id": 11, "name": "Title"},
+]
+
+UNSTRUCTURED_CAT = [
```
Comment (mariannaparzych) [inline helper_scripts/mappings.py:15]: this is valid for previous unstructured with 17 classes, should be updated with new mapping: [paragraph, page_number, image, paragraphs_in_image, title, table, paragraphs_in_table, other, page_header, subheading, formulas, page_footer, paragraphs_in_form, checkbox, checkbox_checked, form, radio_button_checked, radio_button, code_snippet,]

File: helper_scripts/mappings.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    {'id': 16, 'name': 'List'},
+    {'id': 17, 'name': 'Checkbox'},
+    {'id': 18, 'name': 'Abstract'},
+    {'id': 19, 'name': 'Formula'},
+    {'id': 20, 'name': 'PageBreak'},
+    {'id': 21, 'name': 'FigureCaption'},
+    {'id': 22, 'name': 'UncategorizedText'},
+]
+
+MINIHOLISTIC_DOCLAYNET_MAP = {
```
Comment (mariannaparzych) [inline helper_scripts/mappings.py:85]: IMO this can be removed

File: src/super_gradients/common/object_names.py [hunk 6]
```diff
@@ -358,6 +359,7 @@ class ConcatenatedTensorFormats:
 class Dataloaders:
     COCO2017_TRAIN = "coco2017_train"
     COCO2017_VAL = "coco2017_val"
+    COCO2017_VAL_DEEPSENSE = "coco2017_val_deepsense"
```
Comment (mariannaparzych) [inline src/super_gradients/common/object_names.py:362]: you prepared new dataloader? I can not see it in this PR?

File: helper_scripts/evaluate_hi_res.py [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+                    document_labels[document_name][page_number] = label["annotations"]
+
+            for page_number, page_labels in document_labels[document_name].items():
+                # labels are iterated from 0
+                page_predictions = [p for p in document_predictions if p["metadata"]["page_number"] == page_number + 1]
+                formatted_predictions = format_predictions(page_predictions, default_detection_class_prob)
+                formatted_labels = format_labels(page_labels)
+
+                width, height = get_width_height(page_predictions, page_labels, document_name, page_number)
+                if width is None or height is None:
```
[the remark this answers is not in the document]
Comment (amaciaszek-dsai) [reply helper_scripts/evaluate_hi_res.py:167]: For one document in the mini-holistic dataset there's a mismatch between size of the document predictions and document labels. I'm skipping it here

File: helper_scripts/mappings.py (hunk 4 above)
Comment (amaciaszek-dsai) [reply helper_scripts/mappings.py:15]: Done. However, I'm not sure what's actually affected now by the change in this mapping. Do you think it's fine to just 'forget' the old version? :thinking:

File: src/super_gradients/common/object_names.py (hunk 6 above)
Comment (amaciaszek-dsai) [reply src/super_gradients/common/object_names.py]: Oops, that's a leftover. Removed

File: helper_scripts/evaluate_hi_res.py (hunk 3 above)
Comment (amaciaszek-dsai) [reply helper_scripts/evaluate_hi_res.py]: Done

File: helper_scripts/evaluate_hi_res.py (hunk 2 above)
Comment (amaciaszek-dsai) [reply helper_scripts/evaluate_hi_res.py]: Done

File: helper_scripts/mappings.py (hunk 5 above)
Comment (amaciaszek-dsai) [reply helper_scripts/mappings.py]: Removed!

**Pull request merged**